### PR TITLE
Extend mix node to allow per channel mixing

### PR DESCRIPTION
**Pull request by friedererdmann**
# Extending `mix` node to allow per channel mixing

The mix node's mix input should allow for all data formats as input and do a per element linear interpolation, not just float. Example given:

```
in1 = Vector2(1,1)
in2 = Vector2(0,0)
mix = Vector2(1,0)
out = Vector2(1,0)
```

For backwards compatibility float mixing remains supported.

Adding node definitions with the extra suffix of the mix data type like `ND_mix_color3_color3`.

Note: So far I've only tested in MaterialX Viewer, I'm still working on some tests and will try to see results in other renderers.

### CODE REVIEW

**Comment by jstone-lucasfilm** [comment]
This proposal looks good to me, thanks @friedererdmann.  I'm CC'ing @dbsmythe and @niklasharrysson for their thoughts.

**Comment by niklasharrysson** [comment]
Looks good for GLSL and OSL. See attached renders from the testsuite. For MDL there is a small change needed, see above.

[mix_nodes.pdf](https://github.com/AcademySoftwareFoundation/MaterialX/files/9646199/mix_nodes.pdf)

File: libraries/stdlib/genmdl/stdlib_genmdl_impl.mtlx [hunk 1]
```diff
@@ -605,10 +605,15 @@
   <!-- <mix> -->
   <implementation name="IM_mix_float_genmdl" nodedef="ND_mix_float" sourcecode="math::lerp({{bg}}, {{fg}}, {{mix}})" target="genmdl" />
   <implementation name="IM_mix_color3_genmdl" nodedef="ND_mix_color3" sourcecode="math::lerp({{bg}}, {{fg}}, {{mix}})" target="genmdl" />
+  <implementation name="IM_mix_color3_color3_genmdl" nodedef="ND_mix_color3_color3" sourcecode="math::lerp({{bg}}, {{fg}}, {{mix}})" target="genmdl" />
   <implementation name="IM_mix_color4_genmdl" nodedef="ND_mix_color4" sourcecode="mx::stdlib::mx_mix_color4({{fg}}, {{bg}}, {{mix}})" target="genmdl" />
+  <implementation name="IM_mix_color4_color4_genmdl" nodedef="ND_mix_color4_color4" sourcecode="math::lerp({{bg}}, {{fg}}, {{mix}})" target="genmdl" />
```
[the remark this answers is not in the document]
Comment (friedererdmann) [reply libraries/stdlib/genmdl/stdlib_genmdl_impl.mtlx]: Thanks - I overlooked this! Pushed the change just now :)

**Review by niklasharrysson — APPROVED** [review]
Looks good! Thanks @friedererdmann

**Comment by friedererdmann** [comment]
> Looks good! Thanks @friedererdmann

Thanks Niklas for your review and feedback! I don't seem to be able to merge myself, will you merge the PR?

**Review by jstone-lucasfilm — APPROVED** [review]

**Pull request merged**